src/main/java/org/alliancegenome/curation_api/base/entity/AuditedObject.java

```diff
@@ -66,4 +57,61 @@ public class AuditedObject extends BaseEntity {
     @Column(columnDefinition = "boolean default false", nullable = false)
     private Boolean obsolete = false;
 
+    
+    
+    
+    
+    
+    
+    
+    
+    
+    ////////////////////////////////////////////////////////////////////////////////
+    ////////////////////////////////////////////////////////////////////////////////
+    
+    // Handle updating the dates
+    // The idea here is if this is a new object and the dates are null these methods will
+    // set the date to "now" before saving on which ever one or both if they are null
+    
+    // If this is an existing object then it HAS to be loaded in order to make an update on it.
+    // In loading the object we save the dateUpdated to previousDateUpdated
+    // 
+    // if the caller does not set a New date then dateUpdated == previousDateUpdated and we
+    // will force an updated date otherwise the caller did set the date and we will use that instead
+    
```

Comment:
@markquintontulloch so does this logic make sense? And it will happen automatically on every save

Reply:
yeah, I think so